src/data.c

```diff
@@ -1516,6 +1516,9 @@ pr_sendfile_t pr_data_sendfile(int retr_fd, off_t *offset, off_t count) {
         session.total_raw_out += len;
 
         return -1;
+      } else { /* (len == 0) */
+	/* unexpected EOF i.e. if the file is truncated while transfer */
+	break;
```

Comment:
Is this behavior specific to the Solaris `sendfile(2)` implementation?

Reply:
I apologize but I have made a mistake in placing the "break". I reworked it but I will need to test it a bit more.

Regarding to your question:
sendfile() on Solaris can return:
 - positive number =< "count" -> number of bytes transferred in this call
 - -1 on error (check errno)
 - 0 on EOF (no more bytes to transfer)
Will never return any negative number different of -1
If ret. value != -1 then ERRNO is undefined

file can be truncated i.e. by:
 cat /dev/null > file